src/UriString.php

```diff
@@ -259,12 +204,38 @@ public static function parse($uri): array
         }
 
         $uri = (string) $uri;
+        $components = self::parseSimply($uri);
+
+        if ($components !== null) {
+            return $components;
+        }
 
+        $parts = self::checkSchemePath($uri);
+
+        return array_merge(
+            self::URI_COMPONENTS,
+            '' === $parts['authority'] ? [] : self::parseAuthority($parts['acontent']),
+            [
+                'path' => $parts['path'],
+                'scheme' => '' === $parts['scheme'] ? null : $parts['scontent'],
+                'query' => '' === $parts['query'] ? null : $parts['qcontent'],
+                'fragment' => '' === $parts['fragment'] ? null : $parts['fcontent'],
+            ]
+        );
+    }
+
+    /**
+     * Parse URI when it can be simply parsed.
+     *
+     * @throws SyntaxError
+     */
+    private static function parseSimply(string $uri): ?array
```

Comment:
What's the added value of declaring this method ? `parseSimply`

Reply:
Same as checkSchemePath I intended to make code more readable.